src/backends/tensorrt/plan_backend.h

```diff
@@ -268,7 +269,10 @@ class PlanBackend : public InferenceBackend {
     std::vector<void*> buffer_bindings_;
 
     size_t total_batch_size_;
-    std::vector<std::unique_ptr<InferenceRequest>>* requests_;
+    uint64_t compute_start_ns_;
+    uint64_t compute_input_end_ns_;
+    uint64_t compute_output_start_ns_;
+    std::shared_ptr<std::vector<std::unique_ptr<InferenceRequest>>> requests_;
```

Comment:
Use in-place or unique_ptr for requests and responses

Reply:
I am using a shared pointer to the new Payload struct as it will be placed on the SyncQueue structure.